ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/jca/XATerminatorUnitTest.java

```diff
@@ -591,4 +597,122 @@ public SubordinateTransaction get() {
             }
         }, executor);
     }
+
+    @Test
+    public void testCommitMid () throws Exception
```

Comment:
It shows how prior to JBTM-2614 an XATerminator::recover call could have overrwriten the SubordinateTransaction in memory which would then have the consequence of not finding the XAR during recovery and getting an xanoresource error message